runtime/libia2/include/ia2.h

```diff
@@ -131,8 +131,11 @@
 #define IA2_AS_PTR(opaque) (opaque).ptr
 
 /// Get an IA2 opaque function pointer for the wrapped version of `func`
-#define IA2_FN(func)                                                           \
-  (typeof(__ia2_##func)) { (void *)&__ia2_##func }
+#ifdef __cplusplus
+#define IA2_FN(func) reinterpret_cast<typeof &func>((void *)&((__ia2_##func).ptr))
```

Comment:
I don't think `IA2_FN` should do this just because it's C++. In C it expands to a struct while in C++ it's expanding to a function pointer. It made sense to do this for Firefox because of the way we're running the rewriter on a subset of the sources, but I think this macro should expand to the same thing in both languages. We could just throw out the opaque function pointers and have both expand to the same thing but that should be a separate change.

Reply:
That's a good point, this is more of a library-only-mode change (or FF-specific) that has to be expressed differently in C++ than in C.

Reply:
Plus that change wasn't even described by the commit message. Removed.